dae/dae/annotation/annotate_vcf.py

```diff
@@ -27,78 +34,150 @@ def configure_argument_parser() -> argparse.ArgumentParser:
+    """Annotate a region from a given input VCF file using a pipeline."""
+    in_file = VariantFile(input_file)
+    update_header(in_file, pipeline)
+    out_file = VariantFile(out_filepath, "w", header=in_file.header)
+    annotation_attributes = pipeline.annotation_schema.names
+    with pipeline.open():
+        for vcf_var in in_file.fetch(*region):
+            # pylint: disable=use-list-literal
+            buffers: List[List] = [list() for _ in annotation_attributes]
 
             if vcf_var.alts is None:
                 logger.info("vcf variant without alternatives: %s", vcf_var)
                 continue
 
             for alt in vcf_var.alts:
-                annotabale = VCFAllele(
-                    vcf_var.chrom, vcf_var.pos, vcf_var.ref, alt)
-                annotation = pipeline.annotate(annotabale)
+                annotation = pipeline.annotate(
+                    VCFAllele(vcf_var.chrom, vcf_var.pos, vcf_var.ref, alt)
+                )
                 for buff, attribute in zip(buffers, annotation_attributes):
-                    buff.append(str(annotation[attribute]))
+                    # TODO Ask what value to use for missing attr
+                    buff.append(str(annotation.get(attribute, "-")))
 
             for attribute, buff in zip(annotation_attributes, buffers):
-                vcf_var.info[attribute] = ",".join(buff)
-            print(str(vcf_var), file=out_file, end="")
-
+                vcf_var.info[attribute] = buff
+            out_file.write(vcf_var)
+    out_file.close()
     in_file.close()
 
-    if args.output != "-":
-        out_file.close()
+
+def combine(input_filepath, pipeline, part_filenames, output_filepath):
+    """Combine annotated region parts into a single VCF file."""
+    input_file = VariantFile(input_filepath)
+    update_header(input_file, pipeline)
+    with VariantFile(output_filepath, "w", header=input_file.header) as out:
+        for part_filename in part_filenames:
+            part_file = VariantFile(part_filename)
+            for rec in part_file.fetch():
+                out.write(rec)
+    input_file.close()
+
+
+def produce_regions(context, contigs, region_size):
+    """Given a region size, produce contig regions to annotate by."""
+    ref_genome = context.get_reference_genome()
+    assert ref_genome is not None
+    contig_lengths = {}
+    unknown_contigs = []
+    for contig in contigs:
+        try:
+            contig_lengths[contig] = ref_genome.get_chrom_length(contig)
+        except ValueError:
+            logger.warning(
+                "Could not find contig %s in reference genome", contig
+            )
+            unknown_contigs.append((contig,))
+    return [
+        (contig, start, start + region_size)
+        for contig, length in contig_lengths.items()
+        for start in range(1, length, region_size)
+    ] + unknown_contigs
+
+
+def produce_part_filenames(input_filepath, regions, work_dir):
```

Comment:
Add unit tests for this function including tests when `input_filepath` is an absolute path.